cmd/scollector/collectors/disk_linux.go

```diff
@@ -156,6 +163,56 @@ func c_iostat_linux() (opentsdb.MultiDataPoint, error) {
 	return md, err
 }
 
+// Check mdadm raid arrays on linux
+// linux.disk.mdadm.failed value:
+//   0 => active
+//   1 => something else, error
+func c_mdadm_linux() (opentsdb.MultiDataPoint, error) {
+	mdstat, err := os.Open("/proc/mdstat")
+	defer mdstat.Close()
+	if err != nil {
+		// there is no mdadm: not really an error
+		return opentsdb.MultiDataPoint{}, nil
+	}
+	return mdadmLinux(mdstat)
+}
+
+// mock our file for testing purpose
+func mdadmLinux(mdStat io.Reader) (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+	var descr = "0: raid is active, 1: bad state (not active)"
```

Comment:
convention is to declare descriptions as constants outside funcs. Also they are full sentences.